src/resolve/resolved-varlink.c

```diff
@@ -516,6 +532,547 @@ static int vl_method_resolve_address(Varlink *link, JsonVariant *parameters, Var
+                                        JSON_BUILD_PAIR("port", JSON_BUILD_UNSIGNED(rr->srv.port)),
+                                        JSON_BUILD_PAIR("hostname", JSON_BUILD_STRING(normalized))));
+        if (r < 0)
+                return r;
+
+        if ((q->flags & SD_RESOLVED_NO_ADDRESS) == 0) {
+                LIST_FOREACH(auxiliary_queries, aux, q->auxiliary_queries) {
+                        DnsResourceRecord *zz;
+                        DnsQuestion *question;
+                        int ifindex;
+
+                        if (aux->state != DNS_TRANSACTION_SUCCESS)
+                                continue;
+                        if (aux->auxiliary_result != 0)
+                                continue;
+
+                        question = dns_query_question_for_protocol(aux, aux->answer_protocol);
+
+                        r = dns_name_equal(dns_question_first_name(question), rr->srv.name);
+                        if (r < 0)
+                                return r;
+                        if (r == 0)
+                                continue;
+
+                        DNS_ANSWER_FOREACH_IFINDEX(zz, ifindex, aux->answer) {
+                                int family;
+                                const void *p;
+
+                                r = dns_question_matches_rr(question, zz, NULL);
+                                if (r < 0)
+                                        return r;
+                                if (r == 0)
+                                        continue;
+
+                                if (zz->key->type == DNS_TYPE_A) {
+                                        family = AF_INET;
+                                        p = &zz->a.in_addr;
+                                } else if (zz->key->type == DNS_TYPE_AAAA) {
+                                        family = AF_INET6;
+                                        p = &zz->aaaa.in6_addr;
+                                } else {
+                                        r = -EAFNOSUPPORT;
+                                        return r;
+                                }
+
+                                r = json_build(&entry,
+                                        JSON_BUILD_OBJECT(
+                                                        JSON_BUILD_PAIR_CONDITION(ifindex > 0, "ifindex", JSON_BUILD_INTEGER(ifindex)),
+                                                        JSON_BUILD_PAIR("family", JSON_BUILD_INTEGER(family)),
+                                                        JSON_BUILD_PAIR("address", JSON_BUILD_BYTE_ARRAY(p, FAMILY_ADDRESS_SIZE(family)))));
+                                if (r < 0)
+                                        return r;
+
+                                r = json_variant_append_array(_addr, entry);
+                                if (r < 0)
+                                        return r;
+                        }
+                }
```

Comment:
this loop exists in almost the exact same way in `vl_method_resolve_hostname_complete()` already. please unify it in a common helper.

Reply:
Updated in the latest commit